src/binmode/legacy4third.c

```diff
@@ -511,16 +513,54 @@ void legacy_protocol(void) {
                 if (binmode_debug) {
                     printf("\r\n>> ");
                 }
+                bool is_read_sig_cmd = false;
+                uint8_t read_sig_byte_inx = 0;
```

Comment:
Could you add a one-line comment about the purpose of this variable?
I see below that it stores the command buffer's third byte.  e.g., command `0x30 0xYY 0xZZ` ... it would store `0xZZ`, and that the check is only done when that command byte is exactly `0x03`.

Is this storing the data transfer length of the command?  (e.g., checking that the client actually requested at least three bytes of data, and thus the third read byte of data is expected to be valid?)

If so, should this be checking if the data transfer length was `>= 0x03`, rather than `== 0x03` below?

(yes, I'm showing my ignorance of the commands ... I don't have the datasheets available at the moment)

Reply:
Found a datasheet ... and your comments also helped.   The `Read Signature Byte` (`0x30`) command has to be sent for each byte of the signature being read.  

The datasheet shows the instruction format as:

```
0011 0000    xxxx xxxx    xxxx xxbb    oooo oooo
```
With `bb` being the address of the signature byte to be read, and `oooo'oooo` being the single byte data-out (result).  So, my ignorance was why my question was nonsensical! 🙂   Thanks for commenting to help folks like me understand what's going on better.